persistence/jdbc/src/main/java/org/infinispan/persistence/jdbc/table/management/SQLiteTableManager.java

```diff
@@ -0,0 +1,34 @@
+package org.infinispan.persistence.jdbc.table.management;
+
+import org.infinispan.persistence.jdbc.configuration.TableManipulationConfiguration;
+import org.infinispan.persistence.jdbc.connectionfactory.ConnectionFactory;
+import org.infinispan.persistence.jdbc.logging.Log;
+import org.infinispan.util.logging.LogFactory;
+
+/**
+ * @author Ryan Emerson
+ */
+class SQLiteTableManager extends AbstractTableManager {
+
+   private static final Log LOG = LogFactory.getLog(SQLiteTableManager.class, Log.class);
+
+   SQLiteTableManager(ConnectionFactory connectionFactory, TableManipulationConfiguration config, DbMetaData metaData) {
+      super(connectionFactory, config, metaData, LOG);
+   }
+
+   @Override
+   public boolean isUpsertSupported() {
+      // OR/ON CONFLICT introduced in 3.8.11
+      return super.isUpsertSupported() && metaData.getMajorVersion() > 2 && metaData.getMinorVersion() > 7;
```

Comment:
This will return false with version 4.0. And since the functionality was not available in version 3.8, you should also check the micro version - or if not possible, avoid supporting any 3.8.x version.

And finally, I would prefer `>=` comparisons, so that you don't need a comment to explain what version you actually want.